kanidmd/src/lib/idm/oauth2.rs

```diff
@@ -146,14 +149,16 @@ pub struct Oauth2ResourceServersWriteTransaction<'a> {
     inner: CowCellWriteTxn<'a, Oauth2RSInner>,
 }
 
-impl TryFrom<Vec<EntrySealedCommitted>> for Oauth2ResourceServers {
-    type Error = OperationError;
-
-    fn try_from(value: Vec<EntrySealedCommitted>) -> Result<Self, Self::Error> {
+impl Oauth2ResourceServers {
+    pub fn try_new(
+        value: Vec<EntrySealedCommitted>,
+        origin: String,
+    ) -> Result<Self, OperationError> {
         let fernet =
             Fernet::new(&Fernet::generate_key()).ok_or(OperationError::CryptographyError)?;
         let oauth2rs = Oauth2ResourceServers {
             inner: CowCell::new(Oauth2RSInner {
+                origin,
```

Comment:
What's this?

Reply:
Origin is the URL origin of this instance, which can be used for domain name checks etc.